src/computeSourceMap.ts

```diff
@@ -33,3 +51,60 @@ export default function computeSourceMap(
     names: [],
   };
 }
+
+function computeSourceColumns(code: string, tokens: Array<Token>): Array<number> {
+  const sourceColumns: Array<number> = new Array(tokens.length);
+  let j = 0;
+  let currentMapping = tokens[j].start;
+  let lineStart = 0;
+  for (let i = 0; i < code.length; i++) {
+    if (i === currentMapping) {
+      sourceColumns[j] = currentMapping - lineStart;
+      currentMapping = tokens[++j].start;
+    }
+    if (code.charCodeAt(i) === charCodes.lineFeed) {
+      lineStart = i + 1;
+    }
+  }
+  return sourceColumns;
+}
+
+function computeDetailedSourceMap(
+  code: string,
+  filePath: string,
+  {compiledFilename}: SourceMapOptions,
+  rawMappings: Array<number | undefined>,
+  sourceColumns: Array<number>,
+): RawSourceMap {
+  const map = new GenMapping({file: compiledFilename});
+  let j = 0;
+  let currentMapping = rawMappings[j];
+  for (; currentMapping === undefined; currentMapping = rawMappings[++j]);
+  let line = 0;
+  let lineStart = 0;
+  if (currentMapping !== lineStart) {
+    maybeAddSegment(map, line, 0, filePath, line, 0);
+  }
+  for (let i = 0; i < code.length; i++) {
+    if (i === currentMapping) {
+      const genColumn = currentMapping - lineStart;
+      const sourceColumn = sourceColumns[j];
+      maybeAddSegment(map, line, genColumn, filePath, line, sourceColumn);
+      for (
+        currentMapping = rawMappings[++j];
+        currentMapping === i || currentMapping === undefined;
+        currentMapping = rawMappings[++j]
+      );
+    }
+    if (code.charCodeAt(i) === charCodes.lineFeed) {
+      line++;
+      lineStart = i + 1;
+      if (currentMapping !== lineStart) {
+        maybeAddSegment(map, line, 0, filePath, line, 0);
+      }
+    }
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Not a big deal to have this ignore comment, but you may be able to avoid it by adding `ignoreRestSiblings` to the rule config in `.eslintrc.js`. https://eslint.org/docs/latest/rules/no-unused-vars#ignorerestsiblings